hw/ip/otp_ctrl/doc/_index.md

```diff
@@ -414,71 +540,130 @@ Since the life cycle partition is the only partition that needs live updates in-
+The algorithmic steps of these functions are explained in more detail below.
 
-#### Scrambling Mode
+#### Scrambling
 
 As illustrated in subfigure a) in the diagram above, the standard 128bit-key PRESENT configuration with 31 rounds is used for scrambling operations.
 The key used for scrambling is a global netlist constant chosen by the silicon creator, and all secret partitions are encrypted using the their own distinct netlist constant.
 Note that the amount of data that is being scrambled is small (160byte = 20 x 64bit blocks) and the scrambled data remains constant.
 Hence, no additional masking or diversification scheme is applied since only a very limited amount of information can be gathered by observing the scrambling operation via side-channels.
 
-#### Digest Calculation Mode
+#### Digest Calculation
 
 The integrity digests used in the [partition checks]({{< relref "#partition-checks" >}}) are computed using a custom [Merkle-Damgard](https://en.wikipedia.org/wiki/Merkle%E2%80%93Damg%C3%A5rd_construction) construction as illustrated in subfigure b).
 At the beginning of the digest calculation the 64bit state is initialized with an initialization vector (IV).
 Then, the data to be digested is split into 128bit chunks, each of which is used as a 128bit key input for updating the 64bit state with 31 PRESENT encryption rounds.
 Chunks that are not aligned with 128bit are padded with zero, and the finalization operation consists of another 31-round encryption pass with a finalization constant.
 Note that both the IV as well as the finalization constant are global netlist constants chosen by the silicon creator.
 
-#### Scrambling Key Derivation Mode
+#### Unlock Token Hashing
+
+The unlock hashing function is used by the life cycle controller to pass the RAW unlock token through a one-way function before performing the actual token comparison with the netlist constant.
+This makes it harder to reverse engineer and break the RAW unlock process since in addition to extracting the hashed RAW unlock token, an attacker would have to invert the one-way function (or find a collision).
+The hash process is similar to the digest calculation, and uses the PRESENT primitive in a Merkle-Damgard construction to reduce the 128bit token into two 64bit hash-halves as illustrated in subfigure c).
 
-The key derivation modes for ephemeral SRAM and static FLASH scrambling keys leverages the digest mechanism as illustrated in subfigure c).
-In particular, the keys for are derived by repeatedly reducing a semi-random 512bit block of data into a 64bit digest.
 
+Due the PRESENT primitive being a 64bit cipher, the hash produced in this manner is essentially 64bit in strength, which would make the two 64bit halves susceptible to brute-force collision attacks if they where computed completely independently.
+To that end, the second 64bit halve is computed in a chained way by using the first 64bit halve as IV input.
```

Comment:
I forget, did we have a cryptographer sign off on this chained 64-bit idea?  It seems OK to me (certainly better than the independent halves) but there could be subtleties here that make it less strong than we hope/think.

Reply:
No we did not. I was hoping that consultancy could help us with that?

@moidx

Reply:
@moidx  might have found a cryptographer to help here...this is a good thing to double check. 

@msfschaffner do you mind just filing an issue to track crypto questions we want answered for otp? maybe there's only 1, but hopefully we don't forget.

Reply:
I spoke to @moidx today, and he is going to contact a cryptographer and track this in his consultancy docs.